pkg/profiler/profiler.go

```diff
@@ -372,69 +374,52 @@ func (p *CgroupProfiler) Run(ctx context.Context) error {
 	}
 }
 
-func (p *CgroupProfiler) profileLoop(ctx context.Context, captureTime time.Time) error {
-	prof := &profile.Profile{
-		SampleType: []*profile.ValueType{{
-			Type: "samples",
-			Unit: "count",
-		}},
-		TimeNanos:     captureTime.UnixNano(),
-		DurationNanos: int64(p.profilingDuration),
-
-		// We sample at 100Hz, which is every 10 Million nanoseconds.
-		PeriodType: &profile.ValueType{
-			Type: "cpu",
-			Unit: "nanoseconds",
-		},
-		Period: 10000000,
-	}
-
-	mapping := maps.NewMapping(p.pidMappingFileCache)
-	kernelMapping := &profile.Mapping{
-		// TODO(kakkoyun): Check if this conflicts with https://github.com/google/pprof/pull/675/files
-		File: "[kernel.kallsyms]",
-	}
-	kernelFunctions := map[uint64]*profile.Function{}
-	userFunctions := map[[2]uint64]*profile.Function{}
+func (p *CgroupProfiler) profileLoop(ctx context.Context, captureTime time.Time) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
```

Comment:
I think I would prefer if we crash and the user notices more drastically to encourage people to open bugs rather than just continue.

Reply:
Happy to remove it. :+1: